### PR TITLE
refactor: support custom gateway cert expiry days.

### LINKED ISSUE #2046: Custom the envoy-gateway cert life time
current `certgen` job creates the cert's lifetime of 365d by default.

### PR DESCRIPTION
**Pull request by qicz**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #2046

### CODE REVIEW

**Review by zirain — APPROVED** [review]

**Review by zirain — APPROVED** [review]

**Comment by zirain** [comment]
@qicz need at least two approval before merged

**Comment by qicz** [comment]
> @qicz need at least two approval before merged

ok

**Pull request merged**
